modules/oracle-xe/src/main/java/org/testcontainers/containers/OracleContainer.java

```diff
@@ -90,6 +90,11 @@ public OracleContainer withPassword(String password) {
         return self();
     }
 
+    @Override
+    public OracleContainer withUrlParam(String paramName, String paramValue) {
```

Comment:
Since some DBs like Oracle don't support URL parameters, but do support `Properties` parameters, I think it is worth considering renaming this method to `withParam()` or `withJdbcParam()`, so that client code doesn't have to call `withUrlParameter()` for some DBs and something like `withPropertiesParam()` for others (assuming that will be also supported in the future). I just think this would be worth considering.

Reply:
I think this is not entirely correct. What do you think about that, @rnorth ? Thx!

Reply:
Sorry to take so long to come back to this PR to break the tie...

I think the current form is reasonable; Oracle does not support JDBC URL parameters (in the normal sense of a URL parameter, anyway). Also, we don't support setting connection properties for the Oracle driver.

It's good to think about the future case of being agnostic of whether the connection params are set in the URL or as properties. However, I think that:

1. the params themselves are so database-specific that it's rarely going to be possible to write them generically, and 
1. trying to support both URL params and properties via one method is going to tie us in knots if we ever actually support pass-through of properties - we're not going to know which params go where.

Reply:
Makes sense. Let's tackle that problem with connection properties separately. From my POV this commit looks good to go!